code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -13,481 +13,374 @@ What are the archived variables for?
+				enabling these checks should be as easy as possible and the fact that they are disabled should be as clear as possible
+			var/list/max_reqs = reaction.max_requirements
+			if((max_reqs["TEMP"] && temp > max_reqs["TEMP"]) \
+			|| (max_reqs["ENER"] && ener > max_reqs["ENER"]))
+				continue
+			for(var/id in max_reqs)
+				if(id == "TEMP" || id == "ENER")
+					continue
+				if(cached_gases[id] && cached_gases[id][MOLES] > max_reqs[id])
 					continue reaction_loop
-
 			//at this point, all requirements for the reaction are satisfied. we can now react()
-
+			*/
 			. |= reaction.react(src, holder)
 			if (. & STOP_REACTIONS)
 				break
-	if(.)
-		garbage_collect()
-
-///Takes the amount of the gas you want to PP as an argument
-///So I don't have to do some hacky switches/defines/magic strings
-///eg:
-///Tox_PP = get_partial_pressure(gas_mixture.toxins)
-///O2_PP = get_partial_pressure(gas_mixture.oxygen)
+*/
+//Takes the amount of the gas you want to PP as an argument
+//So I don't have to do some hacky switches/defines/magic strings
+//eg:
+//Tox_PP = get_partial_pressure(gas_mixture.toxins)
+//O2_PP = get_partial_pressure(gas_mixture.oxygen)
 
 /datum/gas_mixture/proc/get_breath_partial_pressure(gas_pressure)
-	return (gas_pressure * R_IDEAL_GAS_EQUATION * temperature) / BREATH_VOLUME
-///inverse
+	return (gas_pressure * R_IDEAL_GAS_EQUATION * return_temperature()) / BREATH_VOLUME
+//inverse
 /datum/gas_mixture/proc/get_true_breath_pressure(partial_pressure)
-	return (partial_pressure * BREATH_VOLUME) / (R_IDEAL_GAS_EQUATION * temperature)
+	return (partial_pressure * BREATH_VOLUME) / (R_IDEAL_GAS_EQUATION * return_temperature())
 
-///Mathematical proofs:
-/**
+//Mathematical proofs:
+/*
 get_breath_partial_pressure(gas_pp) --> gas_pp/total_moles()*breath_pp = pp
 get_true_breath_pressure(pp) --> gas_pp = pp/breath_pp*total_moles()
-
 10/20*5 = 2.5
 10 = 2.5/5*20
-**/
+*/
+
+/datum/gas_mixture/turf
+
+/*
+/mob/verb/profile_atmos()
+	/world{loop_checks = 0;}
+	var/datum/gas_mixture/A = new
+	var/datum/gas_mixture/B = new
+	A.parse_gas_string("o2=200;n2=800;TEMP=50")
+	B.parse_gas_string("co2=500;plasma=500;TEMP=5000")
+	var/pa
+	var/pb
+	pa = world.tick_usage
+	for(var/I in 1 to 100000)
+		B.transfer_to(A, 1)
+		A.transfer_to(B, 1)
+	pb = world.tick_usage
+	var/total_time = (pb-pa) * world.tick_lag
+	to_chat(src, "Total time (gas transfer): [total_time]ms")
+	to_chat(src, "Operations per second: [100000 / (total_time/1000)]")
+	pa = world.tick_usage
+	for(var/I in 1 to 100000)
+		B.total_moles();
+	pb = world.tick_usage
+	total_time = (pb-pa) * world.tick_lag
+	to_chat(src, "Total time (total_moles): [total_time]ms")
+	to_chat(src, "Operations per second: [100000 / (total_time/1000)]")
+	pa = world.tick_usage
+	for(var/I in 1 to 100000)
+		new /datum/gas_mixture
+	pb = world.tick_usage
+	total_time = (pb-pa) * world.tick_lag
+	to_chat(src, "Total time (new gas mixture): [total_time]ms")
```

Comment:
Why is this commented out?

Reply:
Fuck if I know. Probably just so you have a reference of what it's doing. The other procs don't have this so I'm going to get rid of it.